src/screens/Profile/ForumNotificationsScreen.js

```diff
@@ -0,0 +1,218 @@
+import React from 'react';
+import {
+  View, StyleSheet, ScrollView,
+  AsyncStorage,
+} from 'react-native';
+import {
+  Title, IconButton, ActivityIndicator, Text, Menu, Divider,
+} from 'react-native-paper';
+import auth from '@react-native-firebase/auth';
+import firestore from '@react-native-firebase/firestore';
+import AsyncLock from 'async-lock';
+import Notification from './Notification';
+import DefaultNotifications from './DefaultNotifications';
+
+export default class ForumNotificationsScreen extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { notifications: [], loading: true, visible: false };
+    this.setNotificationsToViewed = this.setNotificationsToViewed.bind(this);
+    this.setButtonVisibleFalse = this.setButtonVisibleFalse.bind(this);
+    this.setButtonVisibleTrue = this.setButtonVisibleTrue.bind(this);
+    this.onClearAsyncNotifications = this.onClearAsyncNotifications.bind(this);
+    this.onViewPress = this.onViewPress.bind(this);
+    this.onClearPress = this.onClearPress.bind(this);
+    this.lock = new AsyncLock();
+  }
+
+  componentDidMount() {
+    /* TODO: Make each notification connected to the userID for who it should be sent to */
+    this.unsubscribeFromAuth = auth().onAuthStateChanged((user) => {
+      if (user) {
+        this.setState({ currentUserID: user.uid });
+      }
+    });
+
+    this.unsubscribeFromFirestore = firestore().collection('profile_notifications')
+      .orderBy('createdAt', 'desc')
+      .onSnapshot(async (snapshot) => {
+        const newNotificationsSnapshot = snapshot.docChanges().filter((change) => change.type === 'added');
+        const newNotificationDocs = newNotificationsSnapshot.map((change) => change.doc);
+        let newNotifications = newNotificationDocs.map((doc) => ({ ...doc.data(), id: doc.id }));
+        const { currentUserID } = this.state;
+
+        /* Timestamps lose their toDate method when we store them in AsyncStorage,
+        so we include it now */
+        newNotifications = newNotifications.map((notification) => {
+          const finalNotification = notification;
```

Comment:
Remove `finalNotification` and just work with `notification`. This assignment doesn't do anything useful because it's a shallow copy.